library/std/src/sys/wasi/mod.rs

```diff
@@ -83,24 +83,82 @@ pub fn decode_error_kind(errno: i32) -> std_io::ErrorKind {
+        //        wasi::ERRNO_MULTIHOP.raw().into() => "reserved",
+        e if e == wasi::ERRNO_NAMETOOLONG.raw().into() => InvalidFilename,
+        e if e == wasi::ERRNO_NETDOWN.raw().into() => NetworkDown,
+        //        wasi::ERRNO_NETRESET.raw().into() => "connection aborted by network",
+        e if e == wasi::ERRNO_NETUNREACH.raw().into() => NetworkUnreachable,
+        //        wasi::ERRNO_NFILE.raw().into() => "too many files open in system",
+        //        wasi::ERRNO_NOBUFS.raw().into() => "no buffer space available",
+        e if e == wasi::ERRNO_NODEV.raw().into() => NotFound,
+        e if e == wasi::ERRNO_NOENT.raw().into() => NotFound,
+        e if e == wasi::ERRNO_NOEXEC.raw().into() => InvalidData,
+        //        wasi::ERRNO_NOLCK.raw().into() => "no locks available",
+        //        wasi::ERRNO_NOLINK.raw().into() => "reserved",
         e if e == wasi::ERRNO_NOMEM.raw().into() => OutOfMemory,
+        //        wasi::ERRNO_NOMSG.raw().into() => "no message of the desired type",
+        //        wasi::ERRNO_NOPROTOOPT.raw().into() => "protocol not available",
+        e if e == wasi::ERRNO_NOSPC.raw().into() => StorageFull,
+        e if e == wasi::ERRNO_NOSYS.raw().into() => Unsupported,
+        e if e == wasi::ERRNO_NOTCONN.raw().into() => NotConnected,
+        e if e == wasi::ERRNO_NOTDIR.raw().into() => NotADirectory,
+        e if e == wasi::ERRNO_NOTEMPTY.raw().into() => DirectoryNotEmpty,
+        //        wasi::ERRNO_NOTRECOVERABLE.raw().into() => "state not recoverable",
+        e if e == wasi::ERRNO_NOTSOCK.raw().into() => InvalidInput,
+        e if e == wasi::ERRNO_NOTSUP.raw().into() => Unsupported,
+        //        wasi::ERRNO_NOTTY.raw().into() => "inappropriate I/O control operation",
+        e if e == wasi::ERRNO_NXIO.raw().into() => NotFound,
+        //        wasi::ERRNO_OVERFLOW.raw().into() => "value too large to be stored in data type",
+        //        wasi::ERRNO_OWNERDEAD.raw().into() => "previous owner died",
+        e if e == wasi::ERRNO_PERM.raw().into() => PermissionDenied,
+        e if e == wasi::ERRNO_PIPE.raw().into() => BrokenPipe,
+        //        wasi::ERRNO_PROTO.raw().into() => "protocol error",
+        e if e == wasi::ERRNO_PROTONOSUPPORT.raw().into() => Unsupported,
+        e if e == wasi::ERRNO_PROTOTYPE.raw().into() => InvalidInput,
+        //        wasi::ERRNO_RANGE.raw().into() => "result too large",
+        e if e == wasi::ERRNO_ROFS.raw().into() => ReadOnlyFilesystem,
+        //        wasi::ERRNO_SPIPE.raw().into() => "invalid seek",
+        e if e == wasi::ERRNO_SRCH.raw().into() => NotFound,
+        //        wasi::ERRNO_STALE.raw().into() => "reserved",
+        e if e == wasi::ERRNO_TIMEDOUT.raw().into() => TimedOut,
+        e if e == wasi::ERRNO_TXTBSY.raw().into() => ResourceBusy,
+        e if e == wasi::ERRNO_XDEV.raw().into() => CrossesDevices,
+        e if e == wasi::ERRNO_NOTCAPABLE.raw().into() => PermissionDenied,
```

Comment:
`e if wasi::ERRNO_CONSTANT.raw().into()` being repeated for every case was okay a long time ago, when it was just a dozen. I understand why you repeated it, here. But please hoist this repetitive guard into a function so we don't have the same ~18 characters approximately 75 times.